cpp/src/io/json/write_json.cu

```diff
@@ -311,15 +311,15 @@ std::unique_ptr<column> struct_to_strings(table_view const& strings_columns,
                                               -> size_type { return idx / tbl.num_columns(); }));
     auto validity_iterator =
       cudf::detail::make_counting_transform_iterator(0, validity_fn{*tbl_device_view});
-    thrust::exclusive_scan_by_key(rmm::exec_policy(stream),
+    thrust::exclusive_scan_by_key(rmm::exec_policy_nosync(stream),
```

Comment:
Can we also change the `thrust::for_each` in L301 to nosync policy?

Reply:
This one's tricky (at least to me). At first I thought we couldn't because `scatter_fn` would go out of scope before `for_each` runs.
But, `for_each` takes `scatter_fn` by value - trivial data members are copied over. Pointers such as `d_strviews.begin()` should also the fine, as the deallocation is done on the same stream as `for_each`. So it will exist for the duration of the kernel execution.

Reply:
L301 already updated as part of another PR.

Reply:
Reference: L301 updated in https://github.com/rapidsai/cudf/pull/17414